app/controllers/api/regions_controller.rb

```diff
@@ -28,16 +28,19 @@ class RegionsController < BaseController
 
     def show
       data = ActiveSupport::JSON.decode(params[:q]).symbolize_keys
+      data[:session_ids] ||= []
 
-      FLOAT_Q_ATTRS.each { |key| data[key] = data[key].to_f if data.key?(key) }
-      INT_Q_ATTRS.each { |key| data[key] = data[key].to_i if data.key?(key) }
-
-      data[:time_from] = data[:time_from] || Time.new(2010).to_i
-      data[:time_to] = data[:time_to] || Time.new(2100).end_of_year.to_i
+      if (data[:session_ids] != [] && Session.find(data[:session_ids].first).fixed?)
+        respond_with FixedRegionInfo.new.call(data)
```

Comment:
`FixedRegionInfo` is used only for fixed sessions and `RegionInfo` only for mobile ones? Have you considered renaming the latter?